### PR TITLE
Handle summary request type

### PR DESCRIPTION
**Pull request by seswarrajan**
Signed-off-by: Eswar Rajan Subramanian [eswar@accuknox.com](mailto:eswar@accuknox.com)

### CODE REVIEW

**Comment by Ankurk99** [comment]
Can you please squash the commits, and I think we are good to merge

File: src/observability/summary.go [hunk 1]
```diff
@@ -11,7 +12,8 @@ func GetSummaryData(request *opb.Request) (*opb.Response, error) {
 	resp := opb.Response{}
 	var err error = nil
 
-	if request.Type == "system" || request.Type == "all" {
+	if strings.Contains(request.Type, "process") || strings.Contains(request.Type, "file") || strings.Contains(request.Type, "network") {
```
Comment (Ankurk99) [inline src/observability/summary.go:15]: can the request.Type be set to "all"? Then this check won't match, right?

File: src/types/observability.go [hunk 2]
```diff
@@ -160,21 +160,14 @@ type SysObsNwData struct {
...
-	DestinationLabel     string
+	SrcPodName           string
+	DestPodName          string
 	DestinationNamespace string
+	DestinationLabel     string
 	Protocol             string
 	Port                 string
 	Status               string
-	Count                int
+	Count                string
```
Comment (Ankurk99) [inline src/types/observability.go:171]: May I know the reason for converting Count to string?

File: src/observability/summary.go (hunk 1 above)
Comment (seswarrajan) [reply src/observability/summary.go:15]: No @Ankurk99 
From karmor summary, I am planning to pass the request type as 
**"process,file,network"**

File: src/types/observability.go (hunk 2 above)
Comment (seswarrajan) [reply src/types/observability.go:171]: I am sending this data to protobuf as a string, hence the basic storage type us changed to string.

File: src/observability/summary.go (hunk 1 above)
Comment (Ankurk99) [reply src/observability/summary.go:15]: As we discussed, the default type will be all, i.e "process,file,network" so the user won't need to pass "all".

**Review by Ankurk99 — APPROVED** [review]
LGTM :+1:

**Review by wazir-ahmed — APPROVED** [review]
LGTM.

**Pull request merged**